lib/pbench/server/filetree.py

```diff
@@ -243,6 +242,59 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def extract(self, path: Path) -> str:
+        """
+        Extract a file from the tarball and return it as a string
+
+        TODO: Should we allow fetching a full directory path, e.g., "/" to
+        extract everything? And, if so, in what form would we return it.
+        Similarly, what about a binary file?
+
+        Args:
+            path: relative path within the tarball of a file
+
+        Raises:
+            MetadataError if an exception occurs unpacking the tarball
+
+        Returns:
+            The named file as a string
+        """
+        try:
+            return (
+                tarfile.open(self.tarball_path, "r:*").extractfile(path).read().decode()
+            )
+        except Exception as exc:
+            raise MetadataError(self.tarball_path, exc)
+
+    def get_metadata(self) -> JSON:
+        """
+        Fetch the values in metadata.log from the tarball, and return a JSON
+        document organizing the metadata by section.
+
+        TODO: Right now this returns just a few particular items that stand
+        out as "likely useful" ... it may make sense to essentially JSONify the
+        entire ConfigParser payload, although there's a lot of data there
+        that's unlikely to be widely useful.
+
+        Returns:
+            A JSON representation of `metadata.log`
+        """
+        data = self.extract(f"{self.name}/metadata.log")
+        metadata = ConfigParser()
+        metadata.read_string(data)
+        date_str = metadata.get("pbench", "date")
+
+        return {
+            "controller": {"hostname": metadata.get("controller", "hostname")},
+            "pbench": {
+                "config": metadata.get("pbench", "config"),
+                "date": convert_date(date_str, None),
```

Comment:
It seems weird to have part of the rendering of the date done at line 285 and part at line 291.  It strikes me that if you're going to go the trouble of creating a local variable, why not put the final value into it and then just read it at line 291; on the other hand, if it will all fit on one line, then omitting the local variable and "doing it all" at line 291 would be fine by me (it would provide better parallelism for the reader with the surrounding lines).